### ISSUE TITLE
Extend image_flip to cope with other image formats

### ISSUE THREAD

**Issue by fmessmer** [ipa320/cob_perception_common#85]
As found during https://github.com/ipa320/cob_robots/pull/720#pullrequestreview-64362574, `image_flip` currently is only able to flip images that have format `CV_8UC3`

**Comment by fmessmer** [comment]
idea from @ipa-bnm 
additionally, it would make sense to implement image_flip in a lazy matter, i.e. only process and publish messages if there are other nodes sunscribed to the outgoing messages

**Comment by ipa-rmb** [comment]
Yes good idea. If we are lucky, one of my students will have time during the next 2 months to implement these requests.

**Comment by ipa-rmb** [comment]
When .at<type>(v,u) is used for access then we could have a typedef mapping depending on 8,16,32,64 bit, data type (uchar, char, short, ushort, uint, int, int64?, float, double) and channels (1,2,3,4).

We can do something similar for the case with pointers and template/typedef the pointer type (or just use the slower .at operator likewise).
